### ISSUE TITLE
Make properties of KeyboardDragListener mutable

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/scenery-phet#367]
From https://github.com/phetsims/scenery/issues/764 @jonathanolson said

> it's really weird having to create an entirely new KeyboardDragListener every time you want to change one of the deltas or its transform.

It should be possible to change these things.

**Comment by jessegreenberg** [comment]
@zepumph volunteered to work on this, thanks!

**Comment by zepumph** [comment]
I committed getters and setters above. @jessegreenberg please review. 

One thought. I decided that all of these could be mutable because I didn't see anything special happening in the constructor, like linking to the Property, using these properties to set other values. In this manner I feel good about my decision, but it does seem a bit future unproof (¯\_(ツ)_/¯) to do this for a Property, since anyone in the future could just add a link call to it, but not realize there was a setter for it. Perhaps just documentation will be enough?

Oh also, I'm pretty sure that I understand how `bind` works but just to be sure, this line:
`    var stepListener = this.step.bind( this );`

Is safe because `stepListener` is still pointing with a reference to that function which will directly call instance members, rather than saving a "snapshot" of the step function in its current state? If the latter then updating the instance properties won't do anything because the wrapped function isn't pointing to them anymore. Please confirm that it is the former, reference, implementation behind the scenes.

**Comment by pixelzoom** [comment]
Assigned to @jessegreenberg for review for 29 (!) months. Raising priority to high. @jessegreenberg when can you get to this?

**Comment by jessegreenberg** [comment]
Thanks @zepumph, this has been working well and the changes look good. This doesn't need to block publication anymore. But I agree with your concerns around a setter for the Property, I think that one should be removed. Also I don't think the callbacks for `start` `end` and `drag` need to be mutable, and I think it would be better if there weren't getters for them. Looking throughout the project I don't see any usages of them of these setters so I am going to remove them.

**Comment by jessegreenberg** [comment]
Removed in the above commit. I don't see problems fuzzing interactive-description sims.

There was also a question about the binding to `step`, but that listener no longer exists in KeyboardDragListener.

@zepumph back to you to verify you agree with https://github.com/phetsims/scenery/commit/5e9a5e65b991295fffc3c23fffff6d7fff0d5d1e and if there is anything else for this issue.
